utils/utils.c

```diff
@@ -845,3 +849,55 @@ int AlmostEqualRelativeAndAbs_double(double A, double B,
 }
 
 /* #undef __USE_XOPEN2K */
+
+// A parallel cumulative sum
+// Output convention is: cumsum[0] = 0; cumsum[N-1] = sum(a[0:N-1]);
+void parallel_cumsum(const int64_t *a, const int64_t N, int64_t *cumsum){
+    #ifdef _OPENMP
+    int Nthread = omp_get_max_threads();
+    #else
+    int Nthread = 1;
+    #endif
+    
+    int64_t min_N_per_thread = 10000;  // heuristic to avoid overheads
+    if(N/min_N_per_thread < Nthread)
+        Nthread = N/min_N_per_thread;
+    if(Nthread < 1)
+        Nthread = 1;
```

Comment:
Will you please add open and closing braces if the `if` condition is in two lines, otherwise will you please put them in one line